dex/msgjson/types.go

```diff
@@ -463,14 +464,19 @@ func (p *Prefix) Stamp(t, epochIdx, epochDur uint64) {
 // Serialize serializes the Prefix data.
 func (p *Prefix) Serialize() []byte {
 	// serialization: account ID (32) + base asset (4) + quote asset (4) +
-	// order type (1), client time (8), server time (8), epoch ID (16) = 73 bytes
-	b := make([]byte, 0, 73)
+	// order type (1) + client time (8) + server time (8) + epoch index (8) +
+	// epoch duration (8) + commitment (32) = 105 bytes
+	if len(p.AccountID) != 32 || len(p.Commit) != 32 {
+		panic("attempted to serialized uninitialized/invalid Prefix")
+	}
+	b := make([]byte, 0, 105)
 	b = append(b, p.AccountID...)
```

Comment:
Serialize was happy to work with any AccountID, including `Bytes{}` (empty).  This seems problematic.

Reply:
Other than the `Message` constructors, the `msgjson` types perform no data validation, and I don't think they should. This is mostly inline with how the **jsonrpc/types** package works in dcrd, though those types also don't do serialization.

Reply:
I'm certain we don't want panics.  These are here to smoke out bugs (incorrect length data or omitted fields) in tests and elsewhere, and there were a few.  I'm still way about this.  Will convince myself there are no gotchas or make a case for some kind of validation or checks.